reactor/discovery/curate_resource_handler.py

```
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2016-present, CloudZero, Inc. All rights reserved.
+# Licensed under the BSD-style license. See LICENSE file in the project root for full license information.
+
+"""
+Contains the discovery-curate-resource lambda handler.
+Events received are generated by dynamodb streams and assume the following format:
+http://docs.aws.amazon.com/lambda/latest/dg/eventsources.html#eventsources-ddb-update
+"""
+
+import utils.lambda_utils as lambda_utils
+import handlers.decorators as decorators
+import discovery.facade as facade
+import discovery.constants as constants
+import datetime
+import utils
+
+ambient_storage = lambda_utils.AmbientLambdaStorage()
+logger = utils.setup_logging('reactor')
+
+_ENRICH_FUNCTION_TEMPLATE = 'discovery-enrich-{}-resource'
+
+
+@decorators.standard_lambda(lambda: ambient_storage, lambda: logger, log_events=False)
+def curate_resource(event, context):
+    """
+    Receive notifications when resource ID information is added to or removed from the Tracking table.
+    New items are listed in a `pending` state, which means that we need to curate these items and find
+    more information, including discovering the true nature of the resource, its relationships, and perhaps
+    determine if it is a resource at all (sometimes we will get false positives).
+
+    Args:
+        event (dict): An S3->Lambda event, which contains details about the object that was added.
+                      This is expected to contain one or more S3 keys.
+        context: LambdaContext type containing runtime information from the lambda environment
+
+    Returns:
+        None
+    """
+    logger.debug('Resource Event: {}'.format(event))
+
+    for record in event['Records']:
```

Do we need any input validation on `event` here? Consider `event.get('Records', [])` to short circuit the `for` loop without error.

---

I would kind of like it to fail loudly here.  If the event is malformed, which might happen during a manual test, I'd want to know.  Other than that, I have to extend *some* trust to AWS that they will send me the right stuff

---

Fair enough. You've commented it nicely in the docstring. No further questions your honor.